tutorials/Navigation/README.md

```diff
@@ -0,0 +1,222 @@
+4. You want to keep the navigation logic (and probably the business logic) separate from UI.
+
+The first point is quite obvious. If Compose is not the only UI you are using and you want to share the navigation logic, then it can not be managed by Compose.
+
+The second point may be especially useful in Desktop. When a child is pushed to the back stack, it is stopped but not destroyed. So it keeps running in "background" without UI. This makes it possible to keep children's state in memory while navigating.
+
+The third point is about instances retaining, like AndroidX `ViewModels`, and is mostly used in Android. It allows to retain (keep in memory) some data when Android configuration change occurs and the whole navigation stack is recreated. The most important advantage of instance retaining in this pattern is that it is encapsulated in children as implementation details.
+
+The forth point is not that obvious but might be very important. Separating navigation and business logic from the user interface may improve testability. E.g. it becomes possible to test non-UI code in integration with just plain JUnit tests. And the UI can be tested in isolation as well using another testing frameworks.
+
+You can find some integration tests in the TodoApp example:
+
+- [TodoMainTest](https://github.com/JetBrains/compose-jb/blob/master/examples/todoapp/common/main/src/commonTest/kotlin/example/todo/common/main/integration/TodoMainTest.kt) - integration tests for the Main screen.
+- [TodoRootTest](https://github.com/JetBrains/compose-jb/blob/master/examples/todoapp/common/root/src/commonTest/kotlin/example/todo/common/root/integration/TodoRootTest.kt) - integration tests for navigation between the Main and the Edit screens.
+
+This pattern is encouraged by the Decompose library. If this is your choice, then you can just use its recommended approach. 
+
+The main idea is to split (decompose) your project by multiple components. Components can be organized in a tree structure, and each level can (but not must) have multiple `Routers`. Each component is just a normal interface/class, an entry point to the underlying logic.
+
+The only responsibility of the user interface is to listen for components' state changes and trigger their events.
+
+The following resources can help with this pattern:
+- The Decompose [documentation](https://arkivanov.github.io/Decompose/)
+- The [TodoApp](https://github.com/JetBrains/compose-jb/tree/master/examples/todoapp) example
+- The article "[Fully cross-platform Kotlin applications (almost)](https://proandroiddev.com/fully-cross-platform-kotlin-applications-almost-29c7054f8f28)"
+
+### A very basic example:
+
+List child:
+
+``` kotlin
+class List(onItemSelected: (itemId: Long) -> Unit) {
+    // Implementation
+}
+
+@Composable
+fun ListUi(list: List) {
+    // Implementation
```

Comment:
This belongs to the "pattern 1" and this pattern is used in the TodoApp. There is a reference to the example TodoApp at the end of the "pattern 1" section. I think if we include implementation details in the tutorial, it may become outdated quickly.
I can add some implementation details to the "pattern 2" though. 
What do you think?

Reply:
I can add a very basic example of implementations, though

Reply:
> I can add a very basic example of implementations, though

Yes, we can write a simple example that doesn't relate to TodoApp, so it will never be outdated. For example:
```
class List(val onItemSelected: (itemId: Long) -> Unit)

@Composable
fun ListUi(list: List) {
    LazyColumn {
        items(10) {
            Text(
                it.toString(),
                Modifier.clickable {
                    list.onItemSelected(it.toLong())
                }
            )
        }
    }
}


class Details(val itemId: Long, val onFinished: () -> Unit)

@Composable
fun DetailsUi(details: Details) {
    Text(
        details.itemId.toString(),
        Modifier.clickable(onClick = details.onFinished)
    )
}
```